proxy/envoy/config.go

```diff
@@ -354,10 +354,13 @@ func buildHTTPListener(mesh *proxyconfig.MeshConfig, node proxy.Node, instances
 	}
 }
 
-func applyInboundAuth(listener *Listener, mesh *proxyconfig.MeshConfig) {
-	switch mesh.AuthPolicy {
-	case proxyconfig.MeshConfig_NONE:
-	case proxyconfig.MeshConfig_MUTUAL_TLS:
+func shouldApplyAuth(mesh *proxyconfig.MeshConfig, security_option model.SecurityOption) bool {
+	return security_option == model.SecurityEnable ||
+		(mesh.AuthPolicy == proxyconfig.MeshConfig_MUTUAL_TLS && security_option != model.SecurityDisable)
```

Comment:
You don't need the first or condition.. and the second part of the and is just a boolean check